node-notes-cli/command-file.js

```diff
@@ -0,0 +1,119 @@
+import * as fs from 'node:fs/promises';
+
+const command = process.argv[2];
+const input1 = process.argv[3];
+const input2 = process.argv[4];
+if (process.argv.length > 5) {
+  console.error('Please enter up to three arguments. A command followed by two optional arguments.');
+  process.exit(1);
+}
+// valid commands read, create, update, delete
+/**
+ * CLI input to read, add, update and delete notes from a JSON file.
+ */
+switch (command) {
+  case 'read':
+    read();
+    break;
+  case 'create':
+    create();
+    break;
+  case 'update':
+    update();
+    break;
+  case 'delete':
+    deleteEntry();
+    break;
+  default:
+    console.error('Improper command given');
+    process.exit(1);
+}
+
+/**
+ * The read function is the first step for all other functions but
```

Comment:
This double behavior may not be the best choice. It's usually best to have a function have just one single focus. With several variations in behavior, it can be confusing how to make the call, or it may be made mistakenly.